src/core/thread/mle_router.cpp

```diff
@@ -3260,12 +3253,15 @@ otError MleRouter::SendDataResponse(const Ip6::Address &aDestination,
 
 exit:
 
-    if (error != OT_ERROR_NONE && message != NULL)
+    if (error != OT_ERROR_NONE)
     {
-        message->Free();
-    }
+        otLogWarnMle("Failed to send Data Response: %s", otThreadErrorToString(error));
```

Comment:
Can we further reduce the code size by using same formatting string for all send failures?
```suggestion
        otLogWarnMle("Failed to send %s: %s", "Data Response", otThreadErrorToString(error));
```

Reply:
We can consider this, but probably better for a separate PR that covers all instances that are not within scope of this PR.